kong/plugins/key-auth/handler.lua

```diff
@@ -105,6 +106,13 @@ local function unauthorized(message, www_auth_content)
   return { status = 401, message = message, headers = { ["WWW-Authenticate"] = www_auth_content } }
 end
 
+local function remove_query_key(raw_query, key)
+  local pattern = key .. "=[^&]*&?"
+  local new_query = ngx_re_gsub(raw_query, pattern, "", "oj")
+  new_query = ngx_re_gsub(new_query, "&$", "", "oj")
```

Comment:
any error handling we can add here? ngx.re.sub can return errors: https://github.com/openresty/lua-nginx-module/?tab=readme-ov-file#ngxresub

or we can add it in the calling function itself

```
local new_query, err = remove_query_key(raw_query, name)
if err then
  ..
end
```

Reply:
@gruceo - i believe only malformed regex / replace args will reasonably trigger errors here. what are some other errors?

Reply:
The latest commit handles the error.

Reply:
@battlebyte the second `ngx_re_gsub` is not handling the error